server/middlewares.go

```diff
@@ -56,12 +57,21 @@ func Storage(endpoint, db, collection string) gin.HandlerFunc {
 
 		err := backoff.Retry(func() error {
 			var err error
+
 			adapter, err = mongo.New(endpoint, db, collection)
 			if err != nil {
 				return err
 			}
 			listeners.AddListeners(adapter.Emitter())
 
+			// Configuration step for ensure uniqueness attributes in the storage
+			uniqueAttrs, err := attributes.Unique()
+			if err != nil {
+				return err
+			}
+
+			adapter.SetIndexes(uniqueAttrs)
```

Comment:
@alelb @leodido We cannot ensure indexes at this stage.
We need an installation step, because setting indexes on very large collection could take minutes.

May be we can postpone this task, but IMHO it's mandatory.

Reply:
@leogr note that indexes are built in background mode. So read and write operations are available even if the indexes are not yet available.

Reply:
@alelb It implies that uniqueness may not working until background task is done even the scimd is up and running. Isn't it? That's not good anyway :)